code/datums/mood_events/generic_negative_events.dm

```diff
@@ -516,3 +516,15 @@
 	description = "I didn't want to believe it, but there are people out there that are genuinely evil."
 	mood_change = -4
 	timeout = 1 MINUTES
+
+/datum/mood_event/smoke_in_face
+	description = "Cigarette smoke is disgusting."
+	mood_change = -3
+	timeout = 30 SECONDS
+
+/datum/mood_event/smoke_in_face/add_effects(param)
+	if(HAS_TRAIT(owner, TRAIT_ANOSMIA))
```

Comment:
should there be a deviant tastes check here too, or is this too distant from eating/drinking?

Reply:
Nooooooooo

Reply:
~~...maybe smokers? Deviant tastes would be far too uuuuh, Skyrat?~~